### PR TITLE
docs: add 0.12.3 release notes

### PR DESCRIPTION
**Pull request by rb-determined-ai**

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -6,6 +6,49 @@ Release Notes
...
+Version 0.12.3
+^^^^^^^^^^^^^^
+
+**Release Date:** April 27, 2020
+
+- Add a tutorial for the new (experimental) Native API.
+
+- Add support for locally testing experiments via ``det e create --local``.
+
+- Add :class:`determined.experimental.Determined` class for accessing Experiment, Trial, and Checkpoint objects.
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:18]: Can we add links to the appropriate documentation (if it exists for the second and third point, at least)?

File: docs/release-notes.txt [hunk 2]
```diff
@@ -6,6 +6,49 @@ Release Notes
...
+
+- Prevent the ``det trial logs -f`` command from waiting for too long after the trial being watched reaches a terminal state.
+
+- Fix bug where logs disappear when an image is pulled.
+
+- Fix bug that affected the use of :class:`LRScheduler <determined.pytorch.LRScheduler>` in :class:`PyTorchTrial <determined.pytorch.PyTorchTrial>` for multi-GPU training.
+
+- Fix bug after master restart where some errored experiments would show progress indicators.
+
+- Fix bug that stops loading user script if user script exits the system.
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:42]: I had a hard time with this one, it doesn't read correctly but I'm not sure what it's trying to say. Could you get some clarification here?

File: docs/release-notes.txt [hunk 3]
```diff
@@ -6,6 +6,49 @@ Release Notes
...
+
+- TensorBoard logs now appear under the ``storage_path`` for ``shared_fs`` checkpoint configurations.
+
+- Allow commands, notebooks, shells, and TensorBoards to be killed before they are scheduled.
+
+- Print container exit reason in trial logs.
+
+- Choose a better default for the ``--tail`` option of command logs.
+
+- Add additional REST API endpoints for trials.
```
Comment (vishnu2kmohan) [inline docs/release-notes.txt:28]: `additional` feels redundant?
How about `Add REST API endpoints for trials`?

File: docs/release-notes.txt [hunk 4]
```diff
@@ -6,6 +6,49 @@ Release Notes
...
+
+- Support a startup script inside agent docker images.
```
Comment (vishnu2kmohan) [inline docs/release-notes.txt:30]: That :point_up: reads like the startup script is added to a Docker image, which I don't believe is what [pull/243](https://github.com/determined-ai/determined/pull/243) accomplishes.
How about: `Support the execution of a startup script inside the agent docker container`?

File: docs/release-notes.txt [hunk 5]
```diff
@@ -6,6 +6,49 @@ Release Notes
...
+
+- Fix bug where logs disappear when an image is pulled.
+
+- Fix bug that affected the use of :class:`LRScheduler <determined.pytorch.LRScheduler>` in :class:`PyTorchTrial <determined.pytorch.PyTorchTrial>` for multi-GPU training.
+
+- Fix bug after master restart where some errored experiments would show progress indicators.
+
+- Fix bug that stops loading user script if user script exits the system.
+
+- Fix ordering bug of steps from ``det t describe --json``.
```
Comment (vishnu2kmohan) [inline docs/release-notes.txt:44]: How about:
```Fix ordering of steps in the output of ``det t describe --json``.```

File: docs/release-notes.txt (hunk 2 above)
Comment (vishnu2kmohan) [reply docs/release-notes.txt:42]: :heavy_plus_sign: Unclear what the bug (or fix) actually is?

Comment (justin-determined-ai) [reply docs/release-notes.txt:42]: "this one" being "loading user script"

Comment (rb-determined-ai) [reply docs/release-notes.txt:42]: let's just remove this one, I'm pretty sure it has only affected our SEs during dogfooding.

File: docs/release-notes.txt (hunk 3 above)
Comment (rb-determined-ai) [reply docs/release-notes.txt:28]: done

File: docs/release-notes.txt (hunk 4 above)
Comment (rb-determined-ai) [reply docs/release-notes.txt:30]: done

File: docs/release-notes.txt (hunk 5 above)
Comment (rb-determined-ai) [reply docs/release-notes.txt:44]: done

File: docs/release-notes.txt (hunk 1 above)
Comment (rb-determined-ai) [reply docs/release-notes.txt:18]: sure

**Pull request merged**